src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSupport.scala

```diff
@@ -62,4 +62,20 @@ trait LibraryServiceSupport {
     rawlsDAO.getGroupsForUser map {FireCloudConfig.ElasticSearch.discoverGroupNames intersect _}
   }
 
+  def updateAccess(docs: LibrarySearchResponse, workspaces: Seq[WorkspaceListResponse]) = {
+    val updatedResults = docs.results.map { document =>
+      val docId = document.asJsObject.fields.get("workspaceId")
+      val newJson = docId match {
+        case Some(id: JsString) => workspaces.find((workspaceResponse: WorkspaceListResponse) => workspaceResponse.workspace.workspaceId == id.value) match {
+          case Some(workspaceResponse) => document.asJsObject.fields + ("workspaceAccess" -> JsString(workspaceResponse.accessLevel.toString))
+          case _ => document.asJsObject.fields + ("workspaceAccess" -> JsString(WorkspaceAccessLevels.NoAccess.toString))  // or just document.asJsObject.fields ??
+        }
+        case _ => document.asJsObject.fields
+      }
+      JsObject(newJson)
+    }
+
+    docs.copy(results = updatedResults)
+  }
```

Comment:
two comments:
1) at line 71, if you can't find the workspace id in the rawls response, I think you did the right thing and set it to NoAccess.
2) at line 69, `workspaces.find` (i.e. `Seq.find`) is, I believe, an O(n) operation. In the most common use case where you have 20 results from Library, you're executing O(n) 20 times. In this situation, I would create an intermediate `Map` and do your lookups into it, something like:

```
    val workspaceAccess = (workspaces map {w:WorkspaceListResponse => w.workspace.workspaceId -> w.accessLevel}).toMap

    val updatedResults = docs.results.map { document =>
      val docId = document.asJsObject.fields.get("workspaceId")
      val newJson = docId match {
        case Some(id: JsString) => workspaceAccess.get(id.value) match {
          case Some(level) => document.asJsObject.fields + ("workspaceAccess" -> JsString(level.toString))
          case _ => document.asJsObject.fields + ("workspaceAccess" -> JsString(WorkspaceAccessLevels.NoAccess.toString))
        }
        case _ => document.asJsObject.fields
      }
      JsObject(newJson)
    }
```